File: FurnitureStore/FurnitureStore.Models/CustomFurnitureReservation/CustomFurnitureReservation.cs

using FurnitureStore.Models.Account;

namespace FurnitureStore.Models.CustomFurnitureReservation
{
    public class CustomFurnitureReservation
    {
        public long? Id { get; set; }
        public string Note { get; set; }
        public DateTime ReservationDate { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        public bool ReservationStatus { get; set; }
        public string UserId { get; set; }
        //public UserResponse User { get; set; }
    }
}

File: FurnitureStore/UI/furniturestore_admin/lib/models/custom_furniture_reservation/custom_furniture_reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'custom_furniture_reservation.g.dart';

@JsonSerializable()
class CustomFurnitureReservationModel {
  int? id;
  String? note;
  DateTime? reservationDate;
  DateTime? createdDate;
  bool? reservationStatus;
  String? userId;
  CustomFurnitureReservationModel({
    this.id,
    this.note,
    this.reservationDate,
    this.createdDate,
    this.reservationStatus,
    this.userId,
  });

  factory CustomFurnitureReservationModel.fromJson(Map<String, dynamic> json) =>
      _$CustomFurnitureReservationModelFromJson(json);

  Map<String, dynamic> toJson() =>
      _$CustomFurnitureReservationModelToJson(this);
}
